packages/frontend/web/src/pages/Favorites.tsx

```diff
@@ -0,0 +1,87 @@
+import { Link, useNavigate } from 'react-router-dom';
+
+import type { Cocktail } from '@app/types';
+
+import FavoriteHeart from '@/components/FavoriteHeart';
+import Stars from '@/components/favorite/Stars';
+import Title from '@/components/favorite/Title';
+import useFetch from '@/hooks/use-fetch';
+
+const url = `${import.meta.env.VITE_API_URL}/favorite/`;
+
+const image = (image: string, total_degree: number) => {
+  if (image === null) {
+    return total_degree > 0
+      ? '/placeholder-cocktail.webp'
+      : '/placeholder-cocktail-virgin.webp';
+  } else {
+    return `${import.meta.env.VITE_BACKEND_URL}/${image}`;
+  }
+};
+
+export default function Favorite() {
+  const navigate = useNavigate();
+
+  const { data } =
+    useFetch<
+      Pick<
+        Cocktail,
+        'name' | 'id' | 'image' | 'ratings_average' | 'total_degree'
+      >[]
+    >(url);
+
+  if (
+    data &&
+    'message' in data &&
+    'ok' in data &&
+    data.ok === false &&
+    data.message === 'not connected'
+  ) {
```

Comment:
Pourquoi aussi compliqué ?
`if(!data?.ok && data?.message === 'not connected')`

Reply:
TS: 
![Capture d’écran du 2024-01-30 11-29-41](https://github.com/WildCodeSchool/2023-09-orleans-cda-projet2-groupe-a/assets/107045925/10e79692-b459-4274-91de-98dc6e83af9f)
![Capture d’écran du 2024-01-30 11-29-29](https://github.com/WildCodeSchool/2023-09-orleans-cda-projet2-groupe-a/assets/107045925/cb17be60-0da3-4287-adeb-50c8ea81b73b)

Reply:
Yes, `data` est mal typé
Le code d'avant, c'est de la bidouille pour contourner le typage, donc à ne pas faire